File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.ans = 0
        
    def goodNodes(self, root: TreeNode, maxVal = float('-inf')) -> int:
        if root:
            if root.val >= maxVal:
                maxVal = root.val
                self.ans += 1
            self.goodNodes(root.left, maxVal)
            self.goodNodes(root.right, maxVal)
        return self.ans

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans = 0;
    
    public void helper(TreeNode root, int maxVal) {
        if(root.val >= maxVal) {
            maxVal = root.val;
            ans += 1;
        }
        if(root.left != null)
            helper(root.left, maxVal);
        if(root.right != null) 
            helper(root.right, maxVal);  
    }
    
    public int goodNodes(TreeNode root) {
        helper(root, Integer.MIN_VALUE); 
        return ans;
    }
}